lib/game_teams_manager.rb

```diff
@@ -22,46 +22,46 @@ def best_offense
     team_id = group_by(@game_teams, :team_id, :goals).max_by do |team_id, goals_in_game|
       goals_in_game.map(&:to_i).sum.to_f / (goals_in_game.length)
     end.first
-    @tracker.team_manager.teams.find {|team| team.team_id == team_id}.team_name
+    @tracker.team_manager.team_info(team_id)["team_name"]
```

Comment:
You can remove `.team_manager` from all name lines as `@tracker` already knows `team_info`:
`@tracker.team_info(team_id)["team_name"]`